src/main/java/picocli/CommandLine.java

```diff
@@ -2009,13 +2009,45 @@ private static int handleUnhandled(Exception ex, CommandLine cmd, int defaultExi
      */
     private static String exceptionToColorString(Exception ex, Help.ColorScheme existingColorScheme){
         Help.ColorScheme colorScheme = new Help.ColorScheme.Builder(existingColorScheme).applySystemProperties().build();
-        String newLine = System.getProperty("line.separator");
-        StringBuilder stringBuilder = new StringBuilder();
-        stringBuilder.append(colorScheme.errorText(ex.toString())).append(newLine);
-        for (StackTraceElement traceElement : ex.getStackTrace()) {
-            stringBuilder.append(colorScheme.stackTraceText("\tat " + traceElement)).append(newLine);
+        StringWriter stringWriter = new ColorSchemedStringWriter(colorScheme);
+        ex.printStackTrace(new PrintWriter(stringWriter));
+        return stringWriter.toString();
+    }
+
+    /**
+     * Extends StringWriter to use ColorScheme. Allows separating
+     * exception messages from stack traces by intercepting write method.
+     */
+    static class ColorSchemedStringWriter extends StringWriter {
+
+        Help.ColorScheme colorScheme;
+
+        public ColorSchemedStringWriter(Help.ColorScheme colorScheme) {
+            this.colorScheme = colorScheme;
+        }
+
+        public ColorSchemedStringWriter(int initialSize, Help.ColorScheme colorScheme) {
+            super(initialSize);
+            this.colorScheme = colorScheme;
         }
-        return stringBuilder.toString();
+
+        @Override
+        public void write(String str, int off, int len) {
+            if(str.startsWith("\t")){
```

Comment:
Code style comment:
There is a space missing between the `if` and the opening brace `(`, and there is also a space missing between the closing brace `)` and the opening curly `{`. 

This is the case for all "if" statements, but not for "for" statements or "else" statements. Please put spaces to adhere to the style of the existing code. :-)

Reply:
Sorry for that. I will run the code formatter in IDEA before committing.